exp/api/v1beta1/gcpmanagedcontrolplane_types.go

```diff
@@ -27,13 +27,241 @@ const (
 	ManagedControlPlaneFinalizer = "gcpmanagedcontrolplane.infrastructure.cluster.x-k8s.io"
 )
 
+// PrivateCluster defines a private Cluster.
+type PrivateCluster struct {
+	// Enabled defines weather a cluster is private or not. If disabled cluster network is public.
+	// +optional
+	Enabled bool `json:"enabled,omitempty"`
+
+	// ControlPlaneCidrBlock is the IP range in CIDR notation to use for the hosted master network. This range must not
+	// overlap with any other ranges in use within the cluster's network. Honored when enabled is true.
+	// +optional
+	ControlPlaneCidrBlock string `json:"controlPlaneCidrBlock,omitempty"`
+
+	// ControlPlaneGlobalAccess is whenever master is accessible globally or not. Honored when enabled is true.
+	// +optional
+	ControlPlaneGlobalAccess bool `json:"controlPlaneGlobalAccess,omitempty"`
+
+	// ControlPlanePrivateEndpoint is whether the master's internal IP address is used as the cluster endpoint.
+	// Honored when enabled is true.
+	// +optional
+	ControlPlanePrivateEndpoint bool `json:"controlPlanePrivateEndpoint,omitempty"`
+
+	// DisableDefaultSNAT is disables cluster default sNAT rules. Honored when enabled is true.
+	// +optional
+	DisableDefaultSNAT bool `json:"disableDefaultSNAT,omitempty"`
+}
+
+// AuthorizedNetworkCidrBlock defines the CIDR block configuration.
+type AuthorizedNetworkCidrBlock struct {
```

Comment:
If someone is specifying a cidr block i'm guessing that at least one of Name or CidrBlock is required.

Reply:
Yes, adding a MinProperties make sense right?
```diff
+ // +kubebuilder:validation:MinProperties=1
// +kubebuilder:validation:MaxProperties=10
// +optional
CidrBlock map[string]string `json:"cidrBlock,omitempty"`
```

Reply:
Thanks @akshay196 👍